### PR TITLE
Dont remove whitespaces from Db strings

### PR DESCRIPTION
**Pull request by arski**
@murataydos @udemy/team-t Not removing whitespaces from DB strings, otherwise they don't match properly..

### CODE REVIEW

**Comment by arski** [comment]
https://trello.com/c/5W2gWK4R/2974-2-test-videos-comments-not-translated-for-japanese

File: lib/compile.js [hunk 1]
```diff
@@ -130,7 +131,7 @@ var Compiler = (function () {
                     msgid = msgid.replace( unconvertedEntityPattern, convertedEntity );
                 }
 
-                if ( removeWhiteSpaces ) {
+                if ( removeWhiteSpaces && !isDBString ) {
```
Comment (jillesme) [inline lib/compile.js]: What's with the weird spaces between the parentheses? That's jQuery's way of doing JavaScript but we definitely don't do it.

Comment (arski) [reply lib/compile.js]: that's just how it was in angular-gettext, it's not our library, we just forked it and sticking to the same silly layout like they have :)

Comment (pajaco) [reply lib/compile.js]: I'd suggest keeping removeWhiteSpaces here and make isDBString a constituent of the value defined in 
https://github.com/udemy/angular-gettext-tools/blob/5a7eb6ad86667970760d9150da42338fa6862412/lib/compile.js#L107

Comment (pajaco) [reply lib/compile.js]: or rather use `this.shouldRemoveWhiteSpaces(item)`
where  `shouldRemoveWhiteSpaces` encapsulates conditions

**Comment by arski** [comment]
@murataydos ping ping ping

**Comment by luarmr** [comment]
LGTM, I didn't find  any pre tag in crowdin

File: test/fixtures/spaces.po [hunk 2]
```diff
@@ -25,3 +25,8 @@ msgid "Bird"
 msgid_plural "Birds"
 msgstr[0] "Oiseau"
 msgstr[1] "Oiseaux"
+
+#: test/fixtures/db.html
+#: Text from DB
+msgid "This           is\n         a DB test"
+msgstr "Ceci est un DB test"
```
Comment (murataydos) [inline test/fixtures/spaces.po:32]: nice test 👍

**Review by murataydos — APPROVED** [review]
You need to change [this line](https://github.com/udemy/grunt-angular-gettext/blob/master/package.json#L48) to use the new version after you merge this branch to master. And [this line](https://github.com/udemy/website-django/blob/master/static/package.json#L59) after merging that one :)

**Comment by arski** [comment]
@murataydos jeez.. could it be any harder to release this??? :)

any thoughts on the discussion inline about the boolean?

File: lib/compile.js (hunk 1 above)
Comment (murataydos) [reply lib/compile.js]: 👍 I think it makes sense to encapsulate the conditions.

**Comment by arski** [comment]
amended, thanks

**Review by murataydos — APPROVED** [review]
go go go

**Pull request merged**